### PR TITLE
feat: support JSON with MeshAccessLog

### PR DESCRIPTION
**Pull request by michaelbeaumont**
For TCP we now always embed log lines as JSON messages for the log streamer. It still supports old style log lines for now.

### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? -- #4979 
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests -- no new ones
- [x] E2E Tests --
- [x] Manual Universal Tests -- none
- [x] Manual Kubernetes Tests --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: app/kuma-dp/pkg/dataplane/accesslogs/streamer.go [hunk 1]
```diff
@@ -105,12 +106,39 @@ func (s *accessLogStreamer) streamAccessLogs(reader *bufio.Reader) (err error) {
...
+
+		type wrappedMessage struct {
+			Address string      `json:"address"`
+			Message interface{} `json:"message"`
 		}
-		address, accessLogMsg := string(parts[0]), parts[1]
+
+		var wrappedMsg wrappedMessage
+		if err := json.Unmarshal(msg, &wrappedMsg); err != nil {
+			parts := bytes.SplitN(msg, []byte(";"), 2)
```
Comment (jakubdyszkiewicz) [inline app/kuma-dp/pkg/dataplane/accesslogs/streamer.go:119]: that's backward compatibility, right? can you mark it?

File: test/e2e_env/universal/meshaccesslog/meshaccesslog.go [hunk 2]
```diff
@@ -120,6 +121,59 @@ spec:
...
+			stdout, _, err := sinkDeployment.Exec("", "", "head", "-1", "/nc.out")
+			g.Expect(err).ToNot(HaveOccurred())
+
+			type log struct {
+				Source      string
+				Destination string
+				Start       string
+			}
+			var line log
+			g.Expect(json.Unmarshal([]byte(stdout), &line)).To(Succeed())
```
Comment (jakubdyszkiewicz) [inline test/e2e_env/universal/meshaccesslog/meshaccesslog.go:167]: Will it still work when stdout contains multiple json objects? Envoy many times just flushes many logs at once. Shouldn't we try to split by `\n` and take the first item?

Comment (michaelbeaumont) [reply test/e2e_env/universal/meshaccesslog/meshaccesslog.go:167]: Do you mean the e2e test specifically? We call `head` to get one line here.

Comment (jakubdyszkiewicz) [reply test/e2e_env/universal/meshaccesslog/meshaccesslog.go:167]: my bad, didn't noticed `head`, I thought you call `cat`. Thanks!

**Review by jakubdyszkiewicz — APPROVED** [review]

**Pull request merged**